### PR TITLE
iOS Smoke Test (e2e testing)

### PR DESCRIPTION
**Pull request by delianides**
A very basic PR for running detox with rock-native. Right now detox really doesn't like being run with anything less than 7.6 but we could push it through babel to get async/await support. I was able to get jest to connect with detox but was having issues with exports. `detox.init` is supposed to include the expectation library but since jest has its own which is also exported under `expect` it doesn't seem to work. 

You can get this working with mocha by following these steps:
- make sure you're running node 7.6 `nvm use/install 7.6`
- install `fbsimctl`, `brew tap facebook/fb`
`export CODE_SIGNING_REQUIRED=NO && brew install fbsimctl --HEAD`
- run `npm i` to get mocha and detox
- install the detox-cli tools `npm install -g detox-cli`
- start the react-native server `yarn start`
- start the graphql-lambda server `yarn server`
- run `detox build && detox test`

I didn't have any issues running the project on 7.6. 

Notes:
- Follow [detox#143](https://github.com/wix/detox/issues/143) for more information on running jest with detox.
- Initial android support was added as PR today with [detox#148](https://github.com/wix/detox/pull/148)

### CODE REVIEW

**Comment by newspringweb** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please assign someone to merge this PR, and optionally include people who should review.
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  No Changelog changes!
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by delianides** [comment]
FYI these tests will probably fail on CI for right now.

**Review by richarddubay — APPROVED** [review]
I got these tests to run locally. Also updated the snapshots. But the smoke tests fail on CI. I guess we'd need some work to make that happen.

File: e2e/firstTest.spec.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+});
+
+afterAll(async () => {
+  await detox.cleanup();
+});
+*/
+
+describe("Rock Native End-to-End", () => {
+  beforeEach(async () => {
+    await device.reloadReactNative();
```
Comment (jbaxleyiii) [inline e2e/firstTest.spec.js]: dang all of these globals suck

File: e2e/firstTest.spec.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  });
+
+  it('should say hello world', async () => {
+    await expect(element(by.id('mainZone').withDescendant(by.text("Hello World")))).toExist();
+  });
+
+  it('should tap increment', async () => {
+    await element(by.id('increment')).multiTap(2);
```
Comment (jbaxleyiii) [inline e2e/firstTest.spec.js]: that is awesome

**Review by jbaxleyiii — CHANGES_REQUESTED** [review]
@delianides since jest won't work currently, can you setup npm scripts for running these tests? Add the needed dev deps like detox?

We should be able to tell jest to ignore these files until that PR for support is done.

Any idea what CI support looks like?

**Comment by delianides** [comment]
@jbaxleyiii ...and tell jest to ignore the `e2e/` directory?

**Comment by jbaxleyiii** [comment]
@delianides yep!

**Review by JakeDawkins — APPROVED** [review]
The tests have been moved to `/src/__tests__/e2e`. I changed the `yarn e2e` comment to run `detox build && detox test` and configured jest to ignore all `e2e` matching files/folders

**Review by jbaxleyiii — APPROVED** [review]

**Pull request merged**